docs/source/ops/connect/auth0.rst

```diff
@@ -0,0 +1,657 @@
+          return event.user.app_metadata.party;
+        } else {
+          const tokenResponse = await axios.request({
+            "url": "%%ADMIN_TOKEN_URL%%",
+            "method": "post",
+            "data": {
+              "client_id": "%%ADMIN_TOKEN_ID%%",
+              "client_secret": "%%ADMIN_TOKEN_SECRET%%",
+              "audience": "https://daml.com/ledger-api",
+              "grant_type": "client_credentials"
+            },
+            "headers": {
+              "Content-Type": "application/json",
+              "Accept": "application/json"
+            }
+          });
+          const token = tokenResponse.data.access_token;
+          const partyResponse = await axios.request({
+            "url": "%%ORIGIN%%/v1/parties/allocate",
+            "method": "post",
+            "headers": {
+              "Content-Type": "application/json",
+              "Accept": "application/json",
+              "Authorization": "Bearer " + token
+            },
+            "data": {}
+            // only required if JSON API is behind self-signed cert
+            //, httpsAgent: new https.Agent({ rejectUnauthorized: false })
+          });
+          const party = partyResponse.data.result.identifier;
+          api.user.setAppMetadata("party", party);
+
+          // optional one-time setup like creating contracts etc. here
+
+          return party;
+        }
+      };
+      function setToken(party, actAs = [party], readAs = [party], applicationId = event.client.name) {
+        api.idToken.setCustomClaim("https://daml.com/ledger-api", party);
+        api.accessToken.setCustomClaim(
+          "https://daml.com/ledger-api",
+          {
+            "ledgerId": "%%LEDGER_ID%%",
+            "participantId": null,
+            "applicationId": applicationId,
+            "actAs": actAs,
+            "readAs": readAs,
+          });
+      };
+      if (event.client.client_id === "%%LOGIN_ID%%") {
+        const party = await getParty();
+        setToken(party);
+      }
+    };
+
+where you need to replace ``%%LOGIN_ID%%`` with the Client ID of the LOGIN_APP
+application; ``%%ADMIN_TOKEN_URL%%``, ``%%ADMIN_TOKEN_ID%%`` and
+``%%ADMIN_TOKEN_SECRET%%`` with, respectively, the URL, ``client_id`` and
+``client_secret`` values that you can find on the curl example from the Quick
+Start of the ADMIN_TOKEN_APP application; ``%%ORIGIN%%`` by the domain
+(or IP address) and port where Auth0 can reach your JSON API instance; and
+``%%LEDGER_ID%%`` by the ``ledgerId`` you're passing into your Daml driver.
+
+Before we can click on Deploy to save (but not deploy) this snippet, we need to
+do one more thing. This snippet is using a library called ``axios`` to make
+HTTP calls; we need to tell Auth0 about that, so it can provision the library
+at runtime.
+
+To do that, click on the little box icon to the left of code editor, then on
+the button Add Module that that revealed, and type in ``axios`` for the name
```

Comment:
nitpick `that that`
```suggestion
the button Add Module and type in ``axios`` for the name
```

Reply:
I'm pretty sure "that that" is correct here 🤔

Would it sound better if I changed it to "that _this action_ revealed"?